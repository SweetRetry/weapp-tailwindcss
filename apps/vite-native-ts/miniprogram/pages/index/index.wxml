<view class="p-4">
  <IceButton class="bg-[#d72929]" my-class="bg-[orange]"></IceButton>
</view>