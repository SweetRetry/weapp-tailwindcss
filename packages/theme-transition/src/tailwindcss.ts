import plugin from 'tailwindcss/plugin'

export const themeTransition = plugin.withOptions(() => {
  return ({ addBase }) => {
    addBase({

    })
  }
})
