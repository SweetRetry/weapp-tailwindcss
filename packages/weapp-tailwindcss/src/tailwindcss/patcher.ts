import path from 'node:path'
import process from 'node:process'
import type { CacheOptions, ILengthUnitsPatchOptions } from 'tailwindcss-patch'
import { TailwindcssPatcher } from 'tailwindcss-patch'

export function createTailwindcssPatcher(basedir?: string, cacheDir?: string, supportCustomLengthUnitsPatch?: boolean | ILengthUnitsPatchOptions) {
  const cache: CacheOptions = {}

  if (cacheDir) {
    if (path.isAbsolute(cacheDir)) {
      cache.dir = cacheDir
    }
    else if (basedir) {
      cache.dir = path.resolve(basedir, cacheDir)
    }
    else {
      cache.dir = path.resolve(process.cwd(), cacheDir)
    }
  }

  return new TailwindcssPatcher({
    cache,
    patch: {
      basedir,
      applyPatches: {
        exportContext: true,
        extendLengthUnits: supportCustomLengthUnitsPatch,
      },
    },
  })
}
