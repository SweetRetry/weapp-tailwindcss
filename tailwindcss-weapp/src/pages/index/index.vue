<script setup lang="ts">
import { devDependencies } from '../../../package.json'
import BaseLayout from '@/components/BaseLayout.vue'
import FloatButton from '@/components/FloatButton.vue'
import { useSystemStore } from '@/stores'

const version = devDependencies.tailwindcss.slice(1)

const store = useSystemStore()

function openSearchBox() {

}

function toggleTheme() {
  if (store.theme === 'dark') {
    store.setTheme('light')
  }
  else {
    store.setTheme('dark')
  }
}

function go2ThemeDemo() {
  uni.navigateTo({
    url: '/pages/theme/index',
  })
}
</script>

<template>
  <BaseLayout>
    <up-navbar fixed bgColor="inherit" safeAreaInsetTop>
      <template #left>
        <view class="relative flex items-center">
          <view class="i-logos-tailwindcss-icon mr-2 h-[20.57px] w-[34px] shrink-0" />
          <view class="text-lg font-semibold">
            Tailwind CSS
          </view>
          <view
            class="relative bottom-0.5 ml-1 self-end rounded-full bg-sky-400/10 px-2 py-0.5 text-xs font-medium text-sky-600"
          >
            {{ version }}
          </view>
        </view>
      </template>
    </up-navbar>
    <FloatButton store-key="index-float-btn" :padding="[256, 32, 256, 32]">
      <view class="float-btn mb-3">
        <view class="i-mdi-flask-round-bottom-empty-outline text-sky-400 dark:text-sky-500" @click="go2ThemeDemo" />
      </view>
      <view class="float-btn mb-3" @click="toggleTheme">
        <view v-if="store.theme === 'dark'" class="i-mdi-white-balance-sunny text-sky-500" />
        <view v-else class="i-mdi-weather-night text-sky-400" />
      </view>
      <view class="float-btn" @click="openSearchBox">
        <view class="i-mdi-magnify text-sky-400 dark:text-sky-500" />
      </view>
    </FloatButton>
  </BaseLayout>
</template>
