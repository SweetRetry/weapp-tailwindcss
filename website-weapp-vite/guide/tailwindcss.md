# Tailwindcss 集成

相关集成方法，详见:

https://weapp-tw.icebreaker.top/docs/quick-start/native/install
