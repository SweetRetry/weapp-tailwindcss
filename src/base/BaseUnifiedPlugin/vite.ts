import type { Plugin } from 'vite'
import { UserDefinedOptions } from '@/types'
import { getOptions } from '@/defaults'
import type { OutputAsset } from 'rollup'
import { vitePluginName } from '@/constants'
import { getGroupedEntries } from '@/base/shared'
import { getClassCacheSet } from '@/tailwindcss/exposeContext'

// issue 一个节点静态，一个节点动态，动态节点中的静态属性不会被 mangle 导致存在问题

// https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/3
export default function UnifiedViteWeappTailwindcssPlugin(options: UserDefinedOptions = {}): Plugin | undefined {
  const opts = getOptions(options, ['patch', 'style', 'templete', 'js'])
  const { disabled, onEnd, onLoad, onStart, onUpdate, templeteHandler, styleHandler, patch, jsHandler } = opts
  if (disabled) {
    return
  }
  patch?.()

  onLoad()
  // 要在 vite:css 处理之前运行
  return {
    name: vitePluginName,
    enforce: 'post',
    buildStart() {
      onStart()
    },
    generateBundle(opt, bundle, isWrite) {
      // 也许应该都在这里处理
      const entries = Object.entries(bundle).filter(([, s]) => s.type === 'asset') as [string, OutputAsset][]
      const groupedEntries = getGroupedEntries(entries, opts)
      if (Array.isArray(groupedEntries.html)) {
        for (let i = 0; i < groupedEntries.html.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.html[i]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const oldVal = originalSource.source.toString()
          originalSource.source = templeteHandler(oldVal)
          onUpdate(file, oldVal, originalSource.source)
        }
      }
      if (Array.isArray(groupedEntries.css)) {
        for (let i = 0; i < groupedEntries.css.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.css[i]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const rawSource = originalSource.source.toString()
          const css = styleHandler(rawSource, {
            isMainChunk: true
            // classGenerator
          })
          originalSource.source = css
          onUpdate(file, rawSource, css)
        }
      }
      if (Array.isArray(groupedEntries.js)) {
        const set = getClassCacheSet()

        for (let i = 0; i < groupedEntries.js.length; i++) {
          const [file, originalSource] = groupedEntries.js[i]
          const rawSource = originalSource.source.toString()
          const { code } = jsHandler(rawSource, set)
          originalSource.source = code
          onUpdate(file, rawSource, code)
        }
      }
    },
    buildEnd() {
      onEnd()
    }
  }
}
