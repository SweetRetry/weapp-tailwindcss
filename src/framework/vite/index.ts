import type { Plugin } from 'vite'
import { UserDefinedOptions } from '@/types'
import { getOptions } from '@/options'
import type { OutputAsset, OutputChunk } from 'rollup'
import { vitePluginName } from '@/constants'
// import type { Plugin as PostcssPlugin } from 'postcss'
import { getGroupedEntries } from '@/base/shared'
import { createTailwindcssPatcher } from '@/tailwindcss/patcher'

// https://github.com/sonofmagic/weapp-tailwindcss-webpack-plugin/issues/3
export function ViteWeappTailwindcssPlugin(options: UserDefinedOptions = {}): Plugin | undefined {
  const opts = getOptions(options, ['patch', 'style', 'templete'])
  const { disabled, onEnd, onLoad, onStart, onUpdate, templeteHandler, styleHandler, patch } = opts
  if (disabled) {
    return
  }
  patch?.()
  // let globalClassGenerator: ClassGenerator | undefined
  // if (mangle) {
  //   globalClassGenerator = new ClassGenerator(mangle as IMangleOptions)
  // }
  // const cssInjectPreflight = createInjectPreflight(cssPreflight)
  onLoad()
  // 要在 vite:css 处理之前运行
  return {
    name: vitePluginName,
    enforce: 'post',
    buildStart() {
      onStart()
    },
    // configResolved(config) {
    //   // const postcssConfig = config.css?.postcss as {
    //   //   plugins: PostcssPlugin[]
    //   // }
    //   // const tailwindcssIdx = postcssConfig.plugins.findIndex((x) => isRegisterPostcssPlugin(x.postcssPlugin))
    //   // if (tailwindcssIdx === -1) {
    //   //   console.warn('请先安装 tailwindcss! `npm i -D tailwindcss / yarn add -D tailwindcss `')
    //   // }
    //   // const postcssIdx = postcssConfig.plugins.findIndex((x) => x.postcssPlugin === postcssPlugin)
    //   // if (postcssIdx === -1) {
    //   //   postcssConfig.plugins.splice(
    //   //     tailwindcssIdx + 1,
    //   //     0,
    //   //     WeappTailwindcssRenamePlugin({
    //   //       cssMatcher,
    //   //       cssPreflight,
    //   //       mainCssChunkMatcher,
    //   //       replaceUniversalSelectorWith,
    //   //       cssPreflightRange,
    //   //       customRuleCallback
    //   //       // classGenerator
    //   //     }) as PostcssPlugin
    //   //   )
    //   // }
    // },
    // renderChunk (code, chunk, options) {
    //   return code
    // },
    // transform (code, id, options) {
    //   return code
    // },
    generateBundle(opt, bundle, isWrite) {
      // 也许应该都在这里处理
      const entries = Object.entries(bundle).filter(([, s]) => s.type === 'asset') as [string, OutputAsset][]
      const groupedEntries = getGroupedEntries(entries, opts)
      if (Array.isArray(groupedEntries.html)) {
        for (let i = 0; i < groupedEntries.html.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.html[i]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const oldVal = originalSource.source.toString()
          originalSource.source = templeteHandler(oldVal)
          onUpdate(file, oldVal, originalSource.source)
        }
      }
      if (Array.isArray(groupedEntries.css)) {
        for (let i = 0; i < groupedEntries.css.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.css[i]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const rawSource = originalSource.source.toString()
          const css = styleHandler(rawSource, {
            isMainChunk: true
            // classGenerator
          })
          originalSource.source = css
          onUpdate(file, rawSource, css)
        }
      }
    },
    buildEnd() {
      onEnd()
    }
  }
}

export function UnifiedViteWeappTailwindcssPlugin(options: UserDefinedOptions = {}): Plugin | undefined {
  if (typeof options.customReplaceDictionary === 'undefined') {
    options.customReplaceDictionary = 'simple'
  }
  const opts = getOptions(options, ['patch', 'style', 'templete', 'js'])
  const { disabled, onEnd, onLoad, onStart, onUpdate, templeteHandler, styleHandler, patch, jsHandler, mainCssChunkMatcher, appType } = opts
  if (disabled) {
    return
  }

  patch?.()
  const twPatcher = createTailwindcssPatcher()
  onLoad()
  // 要在 vite:css 处理之前运行
  return {
    name: vitePluginName,
    enforce: 'post',
    buildStart() {
      onStart()
    },
    generateBundle(opt, bundle, isWrite) {
      // 也许应该都在这里处理
      // .filter(([, s]) => s.type === 'asset' || s.type === 'chunk')
      const entries = Object.entries(bundle)
      const groupedEntries = getGroupedEntries(entries, opts)
      if (Array.isArray(groupedEntries.html)) {
        for (let i = 0; i < groupedEntries.html.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.html[i] as [string, OutputAsset]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const oldVal = originalSource.source.toString()
          originalSource.source = templeteHandler(oldVal)
          onUpdate(file, oldVal, originalSource.source)
        }
      }
      if (Array.isArray(groupedEntries.css)) {
        for (let i = 0; i < groupedEntries.css.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.css[i] as [string, OutputAsset]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const rawSource = originalSource.source.toString()
          const css = styleHandler(rawSource, {
            isMainChunk: mainCssChunkMatcher(originalSource.fileName, appType)
            // classGenerator
          })
          originalSource.source = css
          onUpdate(file, rawSource, css)
        }
      }
      if (Array.isArray(groupedEntries.js)) {
        const set = twPatcher.getClassSet()

        for (let i = 0; i < groupedEntries.js.length; i++) {
          const [file, originalSource] = groupedEntries.js[i] as [string, OutputChunk]
          const rawSource = originalSource.code
          const { code } = jsHandler(rawSource, set)
          originalSource.code = code
          onUpdate(file, rawSource, code)
        }
      }
    },
    buildEnd() {
      onEnd()
    }
  }
}
