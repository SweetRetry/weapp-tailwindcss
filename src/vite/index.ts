import type { Plugin } from 'vite'
import { UserDefinedOptions } from '@/types'
import { getOptions } from '@/options'
import type { OutputAsset, OutputChunk } from 'rollup'
import { vitePluginName } from '@/constants'
import { getGroupedEntries } from '@/utils'
import { createTailwindcssPatcher } from '@/tailwindcss/patcher'
import { initStore, setRuntimeSet } from '@/mangle/store'

/**
 * @name UnifiedViteWeappTailwindcssPlugin
 * @description uni-app vite vue3 版本插件
 * @link https://weapp-tw.icebreaker.top/docs/quick-start/frameworks/uni-app-vite
 */
export function UnifiedViteWeappTailwindcssPlugin(options: UserDefinedOptions = {}): Plugin | undefined {
  if (typeof options.customReplaceDictionary === 'undefined') {
    options.customReplaceDictionary = 'simple'
  }
  const opts = getOptions(options, ['patch', 'style', 'templete', 'js'])
  const { disabled, onEnd, onLoad, onStart, onUpdate, templeteHandler, styleHandler, patch, jsHandler, mainCssChunkMatcher, appType, mangle } = opts
  if (disabled) {
    return
  }

  patch?.()
  initStore(mangle)
  const twPatcher = createTailwindcssPatcher()
  onLoad()
  // 要在 vite:css 处理之前运行
  return {
    name: vitePluginName,
    enforce: 'post',
    buildStart() {
      onStart()
    },
    generateBundle(opt, bundle, isWrite) {
      // 也许应该都在这里处理
      // .filter(([, s]) => s.type === 'asset' || s.type === 'chunk')
      const entries = Object.entries(bundle)
      const groupedEntries = getGroupedEntries(entries, opts)
      const set = twPatcher.getClassSet()
      setRuntimeSet(set)
      if (Array.isArray(groupedEntries.html)) {
        for (let i = 0; i < groupedEntries.html.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.html[i] as [string, OutputAsset]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const oldVal = originalSource.source.toString()
          originalSource.source = templeteHandler(oldVal)
          onUpdate(file, oldVal, originalSource.source)
        }
      }
      if (Array.isArray(groupedEntries.css)) {
        for (let i = 0; i < groupedEntries.css.length; i++) {
          // let classGenerator
          const [file, originalSource] = groupedEntries.css[i] as [string, OutputAsset]
          // if (globalClassGenerator && globalClassGenerator.isFileIncluded(file)) {
          //   classGenerator = globalClassGenerator
          // }
          const rawSource = originalSource.source.toString()
          const css = styleHandler(rawSource, {
            isMainChunk: mainCssChunkMatcher(originalSource.fileName, appType)
            // classGenerator
          })
          originalSource.source = css
          onUpdate(file, rawSource, css)
        }
      }
      if (Array.isArray(groupedEntries.js)) {
        for (let i = 0; i < groupedEntries.js.length; i++) {
          const [file, originalSource] = groupedEntries.js[i] as [string, OutputChunk]
          const rawSource = originalSource.code
          const { code } = jsHandler(rawSource, set)
          originalSource.code = code
          onUpdate(file, rawSource, code)
        }
      }
    },
    buildEnd() {
      onEnd()
    }
  }
}
