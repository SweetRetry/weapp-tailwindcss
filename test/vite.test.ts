import { describe, it, expect } from 'vitest'
import { build } from 'vite'
import type { RollupOutput } from 'rollup'
import { UnifiedViteWeappTailwindcssPlugin as uvwt } from '@/vite/index'
import path from 'path'

describe('vite test', () => {
  it('vite common build', async () => {
    // 注意： 打包成 h5 和 app 都不需要开启插件配置
    const isH5 = process.env.UNI_PLATFORM === 'h5'
    const isApp = process.env.UNI_PLATFORM === 'app'
    const WeappTailwindcssDisabled = isH5 || isApp

    // vite 插件配置
    const vitePlugins = []
    // postcss 插件配置
    const postcssPlugins = [
      // require('autoprefixer')(),
      require('tailwindcss')({
        config: path.resolve(__dirname, './fixtures/vite/tailwind.config.js')
      })
    ]
    if (!WeappTailwindcssDisabled) {
      vitePlugins.push(uvwt())

      // postcssPlugins.push(
      //   require('postcss-rem-to-responsive-pixel')({
      //     rootValue: 32,
      //     propList: ['*'],
      //     transformUnit: 'rpx'
      //   })
      // )
      // postcssPlugins.push(postcssWeappTailwindcssRename({}))
    }

    const res = (await build({
      root: path.resolve(__dirname, './fixtures/vite/src'),
      plugins: vitePlugins,
      logLevel: 'silent',
      css: {
        postcss: {
          plugins: postcssPlugins
        }
      },
      build: {
        write: false
      }
    })) as RollupOutput

    const output = res.output
    // @ts-ignore
    // output[0].facadeModuleId = switch2relative(output[0].facadeModuleId)
    // Object.keys(output[0].modules).forEach((x) => {
    //   const item = output[0].modules[x]
    //   // @ts-ignore
    //   delete output[0].modules[x].originalLength
    //   if (path.isAbsolute(x)) {
    //     output[0].modules[switch2relative(x)] = item
    //     delete output[0].modules[x]
    //   }
    // })
    expect(output.length).toBe(3)
    expect(output[0].type).toBe('chunk')
    expect(output[0].code).toMatchSnapshot()
    expect(output[1].type).toBe('asset')
    if (output[1].type === 'asset') {
      expect(output[1].source).toMatchSnapshot()
    }
    expect(output[2].type).toBe('asset')
    if (output[2].type === 'asset') {
      expect(output[2].source).toMatchSnapshot()
    }
  })
})
