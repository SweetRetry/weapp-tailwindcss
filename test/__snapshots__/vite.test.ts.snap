// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite test vite common build 1`] = `
Array [
  Object {
    "code": "const s=function(){const t=document.createElement(\\"link\\").relList;if(t&&t.supports&&t.supports(\\"modulepreload\\"))return;for(const e of document.querySelectorAll('link[rel=\\"modulepreload\\"]'))n(e);new MutationObserver(e=>{for(const r of e)if(r.type===\\"childList\\")for(const o of r.addedNodes)o.tagName===\\"LINK\\"&&o.rel===\\"modulepreload\\"&&n(o)}).observe(document,{childList:!0,subtree:!0});function i(e){const r={};return e.integrity&&(r.integrity=e.integrity),e.referrerpolicy&&(r.referrerPolicy=e.referrerpolicy),e.crossorigin===\\"use-credentials\\"?r.credentials=\\"include\\":e.crossorigin===\\"anonymous\\"?r.credentials=\\"omit\\":r.credentials=\\"same-origin\\",r}function n(e){if(e.ep)return;e.ep=!0;const r=i(e);fetch(e.href,r)}};s();
",
    "dynamicImports": Array [],
    "exports": Array [],
    "facadeModuleId": "fixtures/vite/index.html",
    "fileName": "assets/index.1de03ae5.js",
    "implicitlyLoadedBefore": Array [],
    "importedBindings": Object {},
    "imports": Array [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": Object {
      "fixtures/vite/index.css": Object {
        "code": "var index = '';",
        "removedExports": Array [],
        "renderedExports": Array [
          "default",
        ],
        "renderedLength": 15,
      },
      "fixtures/vite/index.html": Object {
        "code": null,
        "removedExports": Array [],
        "renderedExports": Array [],
        "renderedLength": 0,
      },
      "vite/modulepreload-polyfill": Object {
        "code": "const p = function polyfill() {
    const relList = document.createElement('link').relList;
    if (relList && relList.supports && relList.supports('modulepreload')) {
        return;
    }
    for (const link of document.querySelectorAll('link[rel=\\"modulepreload\\"]')) {
        processPreload(link);
    }
    new MutationObserver((mutations) => {
        for (const mutation of mutations) {
            if (mutation.type !== 'childList') {
                continue;
            }
            for (const node of mutation.addedNodes) {
                if (node.tagName === 'LINK' && node.rel === 'modulepreload')
                    processPreload(node);
            }
        }
    }).observe(document, { childList: true, subtree: true });
    function getFetchOpts(script) {
        const fetchOpts = {};
        if (script.integrity)
            fetchOpts.integrity = script.integrity;
        if (script.referrerpolicy)
            fetchOpts.referrerPolicy = script.referrerpolicy;
        if (script.crossorigin === 'use-credentials')
            fetchOpts.credentials = 'include';
        else if (script.crossorigin === 'anonymous')
            fetchOpts.credentials = 'omit';
        else
            fetchOpts.credentials = 'same-origin';
        return fetchOpts;
    }
    function processPreload(link) {
        if (link.ep)
            // ep marker = processed
            return;
        link.ep = true;
        // prepopulate the load record
        const fetchOpts = getFetchOpts(link);
        fetch(link.href, fetchOpts);
    }
};__VITE_IS_MODERN__&&p();",
        "removedExports": Array [],
        "renderedExports": Array [],
        "renderedLength": 1572,
      },
    },
    "name": "index",
    "referencedFiles": Array [],
    "type": "chunk",
    "viteMetadata": Object {
      "importedAssets": Set {},
      "importedCss": Set {
        "assets/index.921729c3.css",
      },
    },
  },
  Object {
    "fileName": "assets/index.921729c3.css",
    "isAsset": true,
    "name": "index.css",
    "source": "::before,::after{--tw-content: \\"\\"}view,:before,:after{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;box-sizing: border-box;border-width: 0;border-style: solid;border-color: currentColor}::-webkit-backdrop{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: }::backdrop{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: }
",
    "type": "asset",
  },
  Object {
    "fileName": "index.html",
    "isAsset": true,
    "name": undefined,
    "source": "<!DOCTYPE html>
<html lang=\\"en\\">

<head>
  <meta charset=\\"UTF-8\\" />
  <script>
    var coverSupport = 'CSS' in window && typeof CSS.supports === 'function' && (CSS.supports('top: env(a)') ||
      CSS.supports('top: constant(a)'))
    document.write(
      '<meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0' +
      (coverSupport ? ', viewport-fit=cover' : '') + '\\" />')
  </script>
  <title></title>
  <!--preload-links-->
  <!--app-context-->
  <script type=\\"module\\" crossorigin src=\\"/assets/index.1de03ae5.js\\"></script>
  <link rel=\\"stylesheet\\" href=\\"/assets/index.921729c3.css\\">
</head>

<body>
  <div id=\\"app\\">
    <!--app-html-->
  </div>
  
</body>

</html>",
    "type": "asset",
  },
]
`;
